controller/src/main/java/org/jboss/as/controller/CapabilityRegistry.java

```diff
@@ -905,6 +909,45 @@ private SatisfactoryCapability findSatisfactoryCapability(String capabilityName,
         return null;
     }
 
+    @Override
+    public Set<String> suggestCapabilities(String referencedCapability,
+            CapabilityScope dependentScope) {
+        if (referencedCapability == null || referencedCapability.isEmpty()
+                || dependentScope == null) {
+            return Collections.emptySet();
+        }
+
+        // Retrieve all the provider points that matching capabilities
+        // must be compliant with. Each ProviderPoint is converted onto a
+        // PathMatcher that is used to filter-in/out the capabilities.
+
+        //For possible capabilities it is always global.
+        CapabilityId id = new CapabilityId(referencedCapability,
```

Comment:
So idea here is to again look trough possible capabilities registry. Not to have completion for actual capabilities already registered.
"Problem" with :get-possible-provider-points operation is that returns possible points that is why there are wildcards in the addresses.